src/Monolog/Logger.php

```diff
@@ -285,9 +285,11 @@ public function addRecord($level, $message, array $context = array())
         foreach ($this->processors as $processor) {
             $record = call_user_func($processor, $record);
         }
-        while (isset($this->handlers[$handlerKey]) &&
```

Comment:
the expected behaviour is documented in the interface:
https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/HandlerInterface.php#L49-L50